roles/module_openapi_cloud/templates/module_directory/cisco_intersight/default_module.j2

```diff
@@ -0,0 +1,331 @@
+    uniquity_keys = []
+{% endif %}
+
+{% if haz_lookup is defined %}
+    _json = None
+    {% if list_index %}
+    if params["{{list_index}}"]:
+        _json = await get_device_info(session, build_url(params), params["{{list_index}}"])
+    {% endif %}
+    if not _json and (uniquity_keys or comp_func):
+        _json = await exists(params, session, url=lookup_url, uniquity_keys=uniquity_keys, per_id_url=per_id_url, comp_func=comp_func)
+    {% if haz_lookup_with_filters is defined %}
+    if not _json:
+        _json = await lookup_with_filters(params, session, build_url(params))
+    {% endif %}
+
+    if _json:
+        if "value" not in _json:  # 7.0.2+
+            _json = {"value": _json}
+        if "_update" in globals():
+            params["{{list_index}}"] = _json["id"]
+            return (await globals()["_update"](params, session))
+
+
+	{% if name == "vcenter_vmtemplate_libraryitems" %}
+        extra_info_url = "https://{api_uri}/api/vcenter/vm-template/library-items/{id}".format(
+            **params,
+            id=_json["id"]
+        )
+        async with session.get(extra_info_url) as resp:
+            if resp.status == 200:
+                extra_json = await resp.json()
+                for k, v in extra_json.items():
+                    _json["value"][k] = v
+	{% endif %}
+
+        return (await update_changed_flag(_json, 200, 'get'))
+{% endif %}
+
+    payload = prepare_payload(params, PAYLOAD_FORMAT["{{ operation }}"])
+    _url = (
+        "https://{api_uri}"
+        "{{ _path }}").format(**params)
+    async with session.{{ verb }}(_url, json=payload, **session_timeout(params)) as resp:
+        if resp.status == 500:
+            text = await resp.text()
+            raise EmbeddedModuleFailure(f"Request has failed: status={resp.status}, {text}")
+        try:
+            if resp.headers["Content-Type"] == "application/json":
+                _json = await resp.json()
+        except KeyError:
+            _json = {}
+
+        if (resp.status in [200, 201]) and "error" not in _json:
+            if isinstance(_json, str): # 7.0.2 and greater
+                _id = _json  # TODO: fetch the object
+            elif isinstance(_json, dict) and "value" not in _json:
+                _id = list(_json["value"].values())[0]
+            elif isinstance(_json, dict) and "value" in _json:
+                _id = _json["value"]
+            _json_device_info = await get_device_info(session, _url, _id)
+            if _json_device_info:
+               _json = _json_device_info
+
+        return await update_changed_flag(_json, resp.status, "{{ operation }}")
+
+
+{% elif operation == "update" -%}
+async def _update(params, session):
+    payload = prepare_payload(params, PAYLOAD_FORMAT["update"])
+    _url = (
+        "https://{api_uri}"
+        "{{ _path }}").format(**params)
+    async with session.get(_url, **session_timeout(params)) as resp:
+        _json = await resp.json()
+        if "value" in _json:
```

Comment:
Can you check if you need all these checks?

Reply:
I've updated with the code for Intersight.